### DISCUSSION TITLE
Can I use this without a Tenant model?

### DISCUSSION THREAD

**Discussion by James4645** [spatie/laravel-multitenancy#366] [Q&A]
I am looking at multitenancy options for a project. I need to switch between a handful of static predefined tenants based on the domain and connect to a different DB for each Tenant. I really don't need Tenant's to be a model, they can be hard coded, and I have no need for a landlord database either, so I rather avoid having to set one up for a single Tenant model with a few never changing rows.

My first thought was just to switch db connection based on domain in custom middleware but I need a way to make jobs and commands work in the context of a tenant so I think this package would solve that but using a full eloquent model is heavy handed for my needs.

**Comment by masterix21** [comment]
Hi, @James4645. I think no because the code is strictly connected to a `Tenant` model.

**Reply by James4645** [reply]
Right it needs a class to get the data about a Tenant, but does it specifically **HAVE** to be an eloquent model or could it be for example a repository class that actually pulls its data from hard coded values rather than SQL queries.
